samples/browser/src/main/java/org/mozilla/samples/browser/BrowserActivity.kt

```diff
@@ -70,4 +90,23 @@ open class BrowserActivity : AppCompatActivity(), ComponentCallbacks2 {
         components.sessionManager.onLowMemory()
         components.icons.onLowMemory()
     }
+
+    private fun observeWebExtensionPopups(): CoroutineScope {
+        return components.store.flowScoped { flow ->
+            flow.ifChanged { it.extensions }
+                .map { it.extensions.filterValues { extension -> extension.popupSession != null } }
```

Comment:
I think here, after the map, we need another empty `.ifChanged()` to make sure that an unrelated change to the global extension state doesn't get collected, otherwise we'd try to display the same popup again.

In other words, we only want to collect if we found a new non-null popupSession.